napi-inl.h

```diff
@@ -6032,12 +6032,12 @@ inline void AsyncProgressWorker<T>::SendProgress_(const T* data, size_t count) {
 
 template <class T>
 inline void AsyncProgressWorker<T>::Signal() const {
-  this->NonBlockingCall(static_cast<T*>(nullptr));
+  this->SendProgress_(static_cast<T*>(nullptr), 0);
```

Comment:
This will override saved progress and make unexpected side effects. We should save a bit of the signal and proceed with null data at https://github.com/nodejs/node-addon-api/blob/main/napi-inl.h#L5998 instead.

Reply:
Hi @legendecas , please take a look at [1f2cd71](https://github.com/nodejs/node-addon-api/pull/1216/commits/1f2cd71a281c393ae7663b3c9513030bf3fe1b5a). Is this what you are talking about with 'save a bit of the signal'?

I also added a test case were an `OnProgress` and `Signal` are performed in sequence, and checks that the progress data is present.

Reply:
Thanks, this looks good!